### PR TITLE
Fix #1569: Allow crossProject's JVM and JS parts to be used in a ProjectRef

### LINKED ISSUE #1569: AssertionError when crossProject is used as a ProjectRef in another project
Hello everyone!

I encountered with a problem appeared when crossProject imports in other one as a ProjectRef.
This repo show how to reproduce this error https://github.com/yelbota/scalajs-crossProjectRef-error
Travis build log https://travis-ci.org/yelbota/scalajs-crossProjectRef-error

### PR DESCRIPTION
**Pull request by yelbota**

### CODE REVIEW

File: sbt-plugin/src/main/scala/scala/scalajs/sbtplugin/cross/CrossType.scala [hunk 1]
```diff
@@ -40,25 +40,23 @@ object CrossType {
 
   object Full extends CrossType {
     def projectDir(crossBase: File, projectType: String): File =
-      crossBase.getAbsoluteFile / projectType
+      crossBase / projectType
```
Comment (sjrd) [inline sbt-plugin/src/main/scala/scala/scalajs/sbtplugin/cross/CrossType.scala]: Restating my comment from the old PR:
@gzm0 What was the reason for the absolute paths in the first place?

**Comment by sjrd** [comment]
test this please

File: sbt-plugin/src/main/scala/scala/scalajs/sbtplugin/cross/CrossType.scala (hunk 1 above)
Comment (gzm0) [inline sbt-plugin/src/main/scala/scala/scalajs/sbtplugin/cross/CrossType.scala]: Well, the project directory should be absolute at this point. We are referring to one physical directory on the filesystem. It seems really strange that removing these fixes anything.

I'll need to take a closer look at this tonight or tomorrow. Please do not merge this unless you know exactly what is going on here.

Comment (yelbota) [inline sbt-plugin/src/main/scala/scala/scalajs/sbtplugin/cross/CrossType.scala]: Some related sbt docs:
http://www.scala-sbt.org/0.13.5/docs/Detailed-Topics/Paths.html#constructing-a-file
http://www.scala-sbt.org/0.13.5/docs/Dormant/Full-Configuration.html#project-references

File: sbt-plugin-test/crossProjectRef/build.sbt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+lazy val crossProjectRef = crossProject.
+  crossType(CrossType.Pure).
+  in(file(".")).
+  settings(scalaVersion := "2.11.6") 
+
+lazy val crossProjectRefJS = crossProjectRef.js
+
+lazy val crossProjectRefJVM = crossProjectRef.jvm
```
Comment (gzm0) [inline sbt-plugin-test/crossProjectRef/build.sbt]: Could you name these differently, so it is clear that they are referenced (and not the reference itself)?

File: sbt-plugin-test/build.sbt [hunk 3]
```diff
@@ -12,8 +12,12 @@ val baseSettings = versionSettings ++ Seq(
     "org.scala-js" %% "scalajs-jasmine-test-framework" % scalaJSVersion % "test"
 )
 
+lazy val crossProjectRefJS = ProjectRef(file("crossProjectRef"), "crossProjectRefJS")
+
+lazy val crossProjectRefJVM = ProjectRef(file("crossProjectRef"), "crossProjectRefJVM")
```
Comment (gzm0) [inline sbt-plugin-test/build.sbt]: Please add a comment explaining what these two things test, also reference the issue number.

Further, consider grouping them together, without an empty line between (what we usually do for JS/JVM pairs).

File: sbt-plugin-test/crossProjectRef/build.sbt [hunk 4]
```diff
@@ -0,0 +1,8 @@
+lazy val crossProjectRef = crossProject.
+  crossType(CrossType.Pure).
+  in(file(".")).
+  settings(scalaVersion := "2.11.6") 
+
+lazy val crossProjectRefJS = crossProjectRef.js
+
```
Comment (gzm0) [inline sbt-plugin-test/crossProjectRef/build.sbt]: Consider not having this empty line.

**Comment by gzm0** [comment]
That's all. Removing the absoluteness makes total sense to me. I really don't know why I added it in the first place.

I can totally see how the reflective loading when you reference a project can run into issues if the pathnames are not relative. IMO it should not, but whether this is a bug in sbt or not is pretty irrelevant at this point, especially since the fix is so easy.

File: sbt-plugin-test/crossProjectRef/build.sbt (hunk 2 above)
Comment (yelbota) [inline sbt-plugin-test/crossProjectRef/build.sbt]: May be "relativeCrossProject"?

Comment (sjrd) [inline sbt-plugin-test/crossProjectRef/build.sbt]: I'd suggest `referencedCrossProject`.

Comment (gzm0) [inline sbt-plugin-test/crossProjectRef/build.sbt]: +1

**Comment by yelbota** [comment]
I have made renamings.

**Comment by sjrd** [comment]
LGTM

**Comment by sjrd** [comment]
test this please

**Comment by sjrd** [comment]
test this please

**Comment by sjrd** [comment]
retest this please

**Comment by sjrd** [comment]
retest this please

**Comment by sjrd** [comment]
retest this please

**Pull request merged**